src/website/views/DashboardContainer.tsx

```diff
@@ -12,41 +13,53 @@ export interface DashboardContainerProps {
   gps: GPSState;
   batteries: BatteriesState;
   windSensors: WindSensorsState;
+  timestamp: TimestampState;
 }
 
 class DashboardContainer extends React.PureComponent<DashboardContainerProps> {
   render() {
     const { gps, batteries, windSensors } = this.props;
 
     const gpsChartData = [
-      gps.data.map((data) => this._parseISOString(data.timestamp)),
+      gps.data
+        .map((data) => this._parseISOString(data.timestamp))
+        .filter((time) => this._validTimestamp(time) == true),
       gps.data.map((data) => data.speed),
     ];
 
     const gpsDistanceData = [
       gps.data.map((data) => data.latitude),
-      gps.data.map((data) => data.longitude)
+      gps.data.map((data) => data.longitude),
     ];
 
     const batteriesVoltageData = [
-      batteries.data.map((data) => this._parseISOString(data.timestamp)),
+      batteries.data
+        .map((data) => this._parseISOString(data.timestamp))
+        .filter((time) => this._validTimestamp(time) == true),
       batteries.data.map((data) => data.batteries[0].voltage),
       batteries.data.map((data) => data.batteries[1].voltage),
     ];
 
     const batteriesCurrentData = [
-      batteries.data.map((data) => this._parseISOString(data.timestamp)),
+      batteries.data
+        .map((data) => this._parseISOString(data.timestamp))
+        .filter((time) => this._validTimestamp(time) == true),
       batteries.data.map((data) => data.batteries[0].current),
       batteries.data.map((data) => data.batteries[1].current),
     ];
 
     const windSensorsSpeedData = [
-      windSensors.data.map((data) => this._parseISOString(data.timestamp)),
+      windSensors.data
+        .map((data) => this._parseISOString(data.timestamp))
+        .filter((time) => this._validTimestamp(time) == true),
       windSensors.data.map((data) => data.windSensors[0].speed),
       windSensors.data.map((data) => data.windSensors[1].speed),
     ];
 
-    const totalTripDistance = this._computeTotalTripDistance(gpsDistanceData[0], gpsDistanceData[1])
+    const totalTripDistance = this._computeTotalTripDistance(
+      gpsDistanceData[0],
+      gpsDistanceData[1],
+    );
```

Comment:
Do we still need to compute the `totalTotalDistance` here? Where is it being used?

Reply:
required for singlevaluechart and singlevalueline when uncommented

Reply:
I saw you uncommeted it. That works, thanks!